tests/jsmarionette/runner/marionette-js-runner/host/python/runner-service/runner_service/handlers/runner.py

```diff
@@ -132,16 +134,62 @@ def __init__(self, *args, **kwargs):
     def start_runner(self, target, options):
         port = options.get('port')
         profile = Profile(profile=options.get('profile'))
-
-        print('---- start runner ----');
-        print(port, profile)
+ 
+        print("---- preparing to start runner ----")
+        print("port =", port, " with profile =", profile)
         self.runner.device.setup_port_forwarding(local_port=port, remote_port=port)
 
+        print("stopping b2g on device before starting runner.")
+        self.runner.device.dm.shellCheckOutput(['stop', 'b2g'])
+
+        # Clean up data on device.
+        print("cleaning up data on device.")
+        data_paths = ['/cache/*',
+                      '/data/local/debug_info_trigger',
+                      '/data/local/indexedDB',
+                      '/data/local/OfflineCache',
+                      '/data/local/permissions.sqlite',
+                      '/data/local/storage/persistent',
+                      '/data/local/storage/default']
+        for path in data_paths:
```

Comment:
Like ideally this would look like

```
def start_runner():
  self._purge_device_data()
  self._forget_wifi_networks()
  self._purge_device_storage()
  self._purge_dirty_webapps()
  // call parent stuff
```